### ISSUE TITLE
Can't find the code where it is saving loss weights during training

### ISSUE THREAD

**Issue by faruknane** [tfzhou/ContrastiveSeg#38]
Hi @tfzhou 

I'm studying your repo for a couple of months from time to time. I'm implementing something different. However, I need to change the loss function and I'm not sure if it will be saved during the training. I couldn't find the lines where the loss function is saved. In init functions, the weights are loaded from configer, but I couldn't find where they are saved. At each epoch, the losses should be saved right? Do I think correctly?

**Comment by oliverester** [comment]
Do you mean you want to save the weights during training? Or, why should you want to save the loss (function)?
If you mean weights: In the _val function, there is the part to store the weights:
segmentor:trainer_constrastive.py:Trainer:__val:

`387  self.module_runner.save_net(self.seg_net, save_mode='performance', experiment=None)
  388  self.module_runner.save_net(self.seg_net, save_mode='val_loss', experiment=None)`

**Comment by faruknane** [comment]
@oliverester I really can't see where the weights are, 
![image](https://user-images.githubusercontent.com/37745467/148407974-276b0901-6889-4687-9f88-559c69339265.png)


Normally, weights are being loaded from configer:
![image](https://user-images.githubusercontent.com/37745467/148408092-9fb1d35d-d8cc-4848-904c-ad36fd4cb31d.png)

I can't see where the weights are written back to the configer. Could you help me please?

**Comment by faruknane** [comment]
@oliverester I also couldn't see any weights in configer.loss after 1 epoch.

![image](https://user-images.githubusercontent.com/37745467/148411773-c470c58a-575f-4543-976c-c372a8524393.png)

**Comment by faruknane** [comment]
@oliverester it seems that the network itself contains three heads that's why it can produce labeled images in testing phase with the given images without needing the loss function. I thought the cross-entropy loss weights were needed
![image](https://user-images.githubusercontent.com/37745467/148421825-295c56b4-8fe7-4e1d-b4d9-1d64578c1749.png)

**Comment by tfzhou** [comment]
What do you mean "loss weights"?

**Comment by faruknane** [comment]
@tfzhou I thought that the last layer is stored in the loss function. However, currently, there are separate heads in the model itself for different loss functions, am I correct? If that's correct, I need to edit the model and add two more heads to do the training.
